### PR TITLE
`Function` - Rename properties and methods to include `prototype`

### LINKED ISSUE #18188: Properties of Function vs. Function.prototype
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

### What specific section or headline is this issue about?

Article title

### What information was incorrect, unhelpful, or incomplete?

The properties of Function instances (`length` and `name`, and I guess the deprecated/nonstandard `arguments`, `caller`, and `displayName`) are named as properties of the Function constructor (e.g., `Function.length`). While it is true that, as a Function itself, the [Function constructor has constant `length` and `name` properties](https://tc39.es/ecma262/#sec-properties-of-the-function-constructor), they are also [present on all Function instances](https://tc39.es/ecma262/#sec-function-instances).

I think it is misleading to title these properties as `Function.length`, etc.—this naming implies that they are present only on the Function constructor rather than Function intsances. It is also inconsistent with how other objects treat their instance properties—e.g., `Array.prototype.length`.

### What did you expect to see?

I expected to see these properties titled `Function.prototype.length`, `Function.prototype.name`, etc. Although `length` and `name` are not strictly speaking inherited from `Function.prototype`, this situation is little different from “exotic” properties on [exotic objects](https://tc39.es/ecma262/#exotic-object), such as [`Array.prototype.length`](https://tc39.es/ecma262/#sec-properties-of-array-instances). It’s less misleading to distinguish them as instance objects using `Function.prototype`.

### Do you have any supporting links, references, or citations?

Links to sections in Ecma262 are given above.

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/function`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/function/index.md
* Last commit: https://github.com/mdn/content/commit/85e526f765521133b837e2465b02bc05432a0934
* Document last modified: 2022-06-27T06:55:54.000Z

</details>

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
Renames properties such as `Function.name` to `Function.prototype.name`.

#### Motivation
Consistency.

#### Supporting details
N/A.

#### Related issues
Fixes #18188.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/function/tostring/index.md [hunk 1]
```diff
@@ -27,9 +27,9 @@ A string representing the source code of the function.
 
 ## Description
 
-The {{jsxref("Function")}} object overrides the {{jsxref("Object.prototype.toString",
-    "toString")}} method inherited from {{jsxref("Object")}}; it does not inherit
-{{jsxref("Object.prototype.toString")}}. For user-defined {{jsxref("Function")}}
+The {{jsxref("Function")}} prototype overrides the {{jsxref("Object.prototype.toString()",
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/function/tostring/index.md:30]: "The Function object" is consistent with how we describe it everywhere else. See also https://github.com/mdn/content/issues/13020

File: files/en-us/web/javascript/reference/global_objects/function/arguments/index.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ The **`function.arguments`** property refers to an array-like object correspondi
 
 ## Description
 
-The syntax `function.arguments` is deprecated. The recommended way to access the {{jsxref("Functions/arguments", "arguments")}} object available within functions is to refer to the variable {{jsxref("Functions/arguments", "arguments")}}.
+The property `function.arguments` is deprecated. The recommended way to access the `arguments` object available within functions is to refer to the variable {{jsxref("Functions/arguments", "arguments")}}.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/function/arguments/index.md:18]: `function.arguments` is _actually_ a syntax. Let's use `someFunction.arguments` instead.

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/arguments/index.md:18]: Hmmm, can we remove the `Function.` prefix where we can? Otherwise `someFunction` or `f` or `fn` or anything else all sound fine to me — whatever fits the context.

File: files/en-us/web/javascript/reference/global_objects/function/call/index.md [hunk 3]
```diff
@@ -112,14 +110,15 @@ const animals = [
   { species: 'Whale', name: 'Fail' }
 ];
 
+function assignPrintMethod(i) {
+  this.print = function() {
+    console.log('#%i %s: %s', i, this.species, this.name);
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/call/index.md:111]: Yes, please

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Overall looks much better to me; mostly minor suggestions

File: files/en-us/web/javascript/reference/global_objects/function/apply/index.md [hunk 4]
```diff
@@ -41,29 +43,29 @@ The result of calling the function with the specified **`this`** value and argum
...
 
 You can assign a different `this` object when calling an existing function. `this` refers to the current object (the calling object). With `apply`, you can write a method once, and then inherit it in another object, without having to rewrite the method for the new object.
 
-`apply` is very similar to {{jsxref("Function.call", "call()")}}, except for the type of arguments it supports. You use an arguments array instead of a list of arguments (parameters). With `apply`, you can also use an array literal, for example, `func.apply(this, ['eat', 'bananas'])`, or an {{jsxref("Array")}} object, for example, `func.apply(this, new Array('eat', 'bananas'))`.
+`apply` is very similar to `call()`, except for the type of arguments it supports. You use an arguments array instead of a list of arguments (parameters). With `apply`, you can also use an array literal, for example, `func.apply(this, ['eat', 'bananas'])`, or an {{jsxref("Array")}} object, for example, `func.apply(this, new Array('eat', 'bananas'))`.
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/apply/index.md]: Hehe, not an issue... These pages need someone to update them anyway.

File: files/en-us/web/javascript/reference/global_objects/function/displayname/index.md [hunk 5]
```diff
@@ -10,13 +10,13 @@ browser-compat: javascript.builtins.Function.displayName
...
 
-The **`function.displayName`** property returns the display name of the function.
+A {{jsxref("Function")}} object's optional **`displayName`** property returns the display name of the function.
 
 ## Examples
 
 ### Setting a displayName
 
-It is usually preferred by consoles and profilers over {{jsxref("Function.name", "func.name")}} to display the name of a function.
+If present, it may be preferred by consoles and profilers over the {{jsxref("Function.prototype.name", "name")}} property, to display the name of a function.
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/displayname/index.md:19]: In that case should be "to be displayed as" :P

**Comment by Josh-Cena** [comment]
No—I'm not dropping random thoughts as I go through the diff. I'm not particularly interested in re-writing _outside_ the diff.

File: files/en-us/web/javascript/reference/global_objects/function/length/index.md [hunk 6]
```diff
@@ -9,21 +9,21 @@ browser-compat: javascript.builtins.Function.length
...
 
 ### Data property of the Function constructor
 
-The {{jsxref("Function")}} constructor is itself a {{jsxref("Function")}} object. Its `length` data property has a value of 1. The property attributes are: Writable: `false`, Enumerable: `false`, Configurable: `true`.
+The {{jsxref("Function")}} constructor is itself a `Function` object. Its `length` data property has a value of `1`. The property attributes are: Writable: `false`, Enumerable: `false`, Configurable: `true`.
 
 ### Property of the Function prototype object
 
-The length property of the {{jsxref("Function")}} prototype object has a value of 0.
+The length property of the {{jsxref("Function")}} prototype object has a value of `0`.
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/length/index.md:26]:
```suggestion
Due to historical reasons, `Function.prototype` is a callable itself. The `length` property of `Function.prototype` has a value of `0`.
```

**Review by Josh-Cena — APPROVED** [review]
I've done a few changes and at least the diffed regions look good enough to me.

I'm going to go ahead and merge this. Any extra modernization attempts can go into follow-up PRs.

File: files/en-us/web/javascript/reference/global_objects/function/bind/index.md [hunk 7]
```diff
@@ -23,28 +23,15 @@ is called.
 
 ```js
 bind(thisArg)
-bind(thisArg, arg1)
-bind(thisArg, arg1, arg2)
-bind(thisArg, arg1, ... , argN)
+bind(thisArg, arg1, …, argN)
```
Comment (OnkarRuikar) [inline files/en-us/web/javascript/reference/global_objects/function/bind/index.md:26]: @Phosra why did you change these?
As per the MDN Web Docs guidelines, https://developer.mozilla.org/en-US/docs/MDN/Structures/Syntax_sections#syntax_for_arbitrary_number_of_parameters , it's correct. Including the ellipsis.

We updated all the pages as per these guidelines couple of month ago.

File: files/en-us/web/javascript/reference/global_objects/function/call/index.md [hunk 8]
```diff
@@ -20,47 +18,31 @@ individually.
 ```js
 call()
 call(thisArg)
-call(thisArg, arg1)
-call(thisArg, arg1, arg2)
-call(thisArg, arg1, ... , argN)
+call(thisArg, arg1, …, argN)
```
Comment (OnkarRuikar) [inline files/en-us/web/javascript/reference/global_objects/function/call/index.md:21]: call(thisArg, arg1, /* … ,*/ argN)

File: files/en-us/web/javascript/reference/global_objects/function/bind/index.md (hunk 7 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/function/bind/index.md:26]: Oh @OnkarRuikar I wasn't aware of that convention ^^ If you want to send a quick fix I will gladly review it.
